docs/template_plugin/src/template_plugin.cpp

```diff
@@ -61,14 +62,13 @@ std::shared_ptr<ngraph::Function> TransformNetwork(const std::shared_ptr<const n
         new_types.emplace_back(parameter->get_element_type());
     }
 
-    auto clonedNetwork = ngraph::specialize_function(std::const_pointer_cast<ngraph::Function>(function),
-        new_types, new_shapes, std::vector<void *>(new_types.size(), nullptr), constFolding, shareConsts);
-
-    auto transformedNetwork = clonedNetwork;
-    transformedNetwork->set_friendly_name(function->get_friendly_name());
+    auto transformedNetwork = ngraph::clone_function(*function);
```

Comment:
Looks like clone_function has not friendly interface, may be we change it to `std::shared_ptr<ngraph::Function> clone_function(const std::shared_ptr<const ngraph::Function>& function)`

Reply:
I propose to do it separately.

Reply:
Ok but lets do not forget to change this